process_report/tests/unit/invoices/test_pi_specific_invoice.py

```diff
@@ -1,68 +1,147 @@
+            .copy()
+            .reset_index(drop=True)
+        )
+        answer_invoice_pi1.loc[len(answer_invoice_pi1)] = None
+        answer_invoice_pi1.loc[
+            answer_invoice_pi1.index[-1], ["Invoice Month", "Balance"]
+        ] = ["Total", 300]
+        for column_name in [
+            "Prepaid Group Balance",
+            "Prepaid Group Used",
+            "Balance",
+        ]:
+            answer_invoice_pi1[column_name] = answer_invoice_pi1[column_name].apply(
+                add_dollar_sign
+            )
+        answer_invoice_pi1.fillna("", inplace=True)
+
+        answer_invoice_pi2 = (
+            test_invoice[test_invoice["Manager (PI)"] == "PI2"]
+            .copy()
+            .reset_index(drop=True)
+        )
+        answer_invoice_pi2.loc[len(answer_invoice_pi2)] = None
+        answer_invoice_pi2.loc[
+            answer_invoice_pi2.index[-1], ["Invoice Month", "Balance"]
+        ] = ["Total", 700]
+        answer_invoice_pi2 = answer_invoice_pi2.drop(
+            [
+                "Prepaid Group Name",
+                "Prepaid Group Institution",
+                "Prepaid Group Balance",
+                "Prepaid Group Used",
+            ],
+            axis=1,
         )
+        answer_invoice_pi2["Balance"] = answer_invoice_pi2["Balance"].apply(
+            add_dollar_sign
+        )
+        answer_invoice_pi2.fillna("", inplace=True)
+
+        pi_inv = test_utils.new_pi_specific_invoice(data=test_invoice)
+        output_invoice = pi_inv._get_pi_dataframe(test_invoice, "PI1")
+        self.assertTrue(answer_invoice_pi1.equals(output_invoice))
+
+        output_invoice = pi_inv._get_pi_dataframe(test_invoice, "PI2")
+        self.assertTrue(answer_invoice_pi2.equals(output_invoice))
 
-        pi_df = pandas.read_csv(output_dir.name + "/" + pi_csv_1)
-        self.assertEqual(len(pi_df["Manager (PI)"].unique()), 1)
-        self.assertEqual(
-            pi_df["Manager (PI)"].unique()[0], self.dataframe["Manager (PI)"][0]
+    @mock.patch("process_report.invoices.invoice.Invoice._filter_columns")
+    @mock.patch("os.path.exists")
+    @mock.patch("subprocess.run")
+    def test_export_pi(self, mock_subprocess_run, mock_path_exists, mock_filter_cols):
+        invoice_month = "2024-10"
+        test_invoice = self._get_test_invoice(
+            ["PI1", "PI1", "PI2", "PI2"],
+            [
+                "BU",
+                "BU",
+                "HU",
+                "HU",
+            ],
+            [100, 200, 300, 400],
+            group_name=[None, "G1", None, None],
         )
 
-        self.assertIn("ProjectA", pi_df["Project - Allocation"].tolist())
-        self.assertIn("ProjectB", pi_df["Project - Allocation"].tolist())
-        self.assertIn("ProjectC", pi_df["Project - Allocation"].tolist())
+        mock_filter_cols.return_value = test_invoice
+        mock_path_exists.return_value = True
+        output_dir = tempfile.TemporaryDirectory()
```

Comment:
We don't need to create a temporary directory -- because we're mocking out `subprocess.run`, we never create any output.

You can do this instead:

```
        pi_inv = test_utils.new_pi_specific_invoice(
            "/fakedir", invoice_month, data=test_invoice
        )
        pi_inv.process()
        pi_inv.export()
        pi_pdf_1 = f"/fakedir/BU_PI1_{invoice_month}.pdf"
        pi_pdf_2 = f"/fakedir/HU_PI2_{invoice_month}.pdf"
```